### PR TITLE
[PM-164] Enable mock psp in uat

### PR DESCRIPTION
**Pull request by infantesimone**
### List of changes

- enable `mock-psp` in uat env
- update `url` and `username` conf for `postgress` in order to set dev env, in any env.
- add flyway migrations for mock-psp db

### Type of changes

- [ ] Add new resources
- [x] Update configuration to existing resources
- [ ] Remove existing resources

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [ ] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [ ] No

### Other information

---

### CODE REVIEW

File: src/env/uat/terraform.tfvars [hunk 1]
```diff
@@ -67,11 +67,12 @@ postgresql_network_rules = {
   # dblink
   allow_access_to_azure_services = false
 }
-prostgresql_db_mockpsp = "mock-psp"
+prostgresql_db_mockpsp    = "mock-psp"
+postgresql_schema_mockpsp = "mock_psp_uat"
```
Comment (pasqualedevita) [inline src/env/uat/terraform.tfvars:71]: Can we call schema mock_psp without env? in this way we reduce differences between envs

**Comment by pasqualedevita** [comment]
also I suggest to migrate db mock to new postgresql_server module that support basic sku
U can copy from here https://github.com/pagopa/userregistry-infra/blob/main/src/pillar/05_db.tf#L102

File: src/env/uat/terraform.tfvars (hunk 1 above)
Comment (infantesimone) [reply src/env/uat/terraform.tfvars:71]: right 👍

Comment (infantesimone) [reply src/env/uat/terraform.tfvars]: ✅

**Comment by infantesimone** [comment]
> also I suggest to migrate db mock to new postgresql_server module that support basic sku U can copy from here https://github.com/pagopa/userregistry-infra/blob/main/src/pillar/05_db.tf#L102

✅

File: src/env/uat/terraform.tfvars (hunk 1 above)
Comment (infantesimone) [reply src/env/uat/terraform.tfvars]: I updated postgress module, and explicitly added `privatelink_postgres_database_azure_com` and `privatelink_postgres_database_azure_com_vnet` (I think there were some changes made on the portal), @pasqualedevita  feedback welcome:-)

**Comment by infantesimone** [comment]
> > also I suggest to migrate db mock to new postgresql_server module that support basic sku U can copy from here https://github.com/pagopa/userregistry-infra/blob/main/src/pillar/05_db.tf#L102
> 
> ✅

I updated postgress module, and explicitly added privatelink_postgres_database_azure_com and privatelink_postgres_database_azure_com_vnet (I think there were some changes made on the portal), @pasqualedevita feedback welcome :-)

**Review by pasqualedevita — COMMENTED** [review]
Can we set postgresql sku to basic in dev? now we can disable private endpoints and use a lower sku for dev envs

**Review by pasqualedevita — DISMISSED** [review]

**Review by uolter — APPROVED** [review]

**Review by uolter — APPROVED** [review]

**Review by uolter — APPROVED** [review]

**Pull request merged**